lib/IAMClient.js

```diff
@@ -449,11 +295,19 @@ class IAMClient {
         const req = this.useHttps ?
             https.request(options) : http.request(options);
 
-        auth.generateV4Headers(req, JSON.stringify(data),
-                               this.accessKey, this.secretKeyValue);
-
         if (method === 'POST' && typeof data === 'object') {
-            req.write(JSON.stringify(data));
+            auth.generateV4Headers(req, data,
+                    this.accessKey, this.secretKeyValue, 'iam');
+            const paramsToSend = [];
+            Object.keys(data).forEach(key => {
```

Comment:
You can use nodejs' `Query String` module for this.
`querystring.stringify` https://nodejs.org/dist/latest-v4.x/docs/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options
It even accepts `encodeURIComponent` as it's options.